packages/visualizations/src/components/Map/utils.js

```diff
@@ -53,65 +52,94 @@ export const mapKeyToColor = (values, colorScale) => {
     const scale = chroma.scale([colorMin, colorMax]).domain([min, max]);
 
     const mapping = {};
-    
+
     values.forEach(v => {
         mapping[v.x] = scale(v.y).hex();
     });
 
     return mapping;
 }
 
-// This is a bound that can only be extended, and will take the value of the first bound that extends it
-const VOID_BOUNDS = [[
-    180, 
+// This is a default bound that will be extended
+const VOID_BOUNDS = [
+    180,
     90,
-], [
-    -180, 
+    -180,
     -90,
-]];
+];
+
+function computeBboxFromCoords(coordsPath, bbox) {
+    return coordsPath.reduce((current, coords) => {
+        return [
+            Math.min(coords[0], current[0]),
+            Math.min(coords[1], current[1]),
+            Math.max(coords[0], current[2]),
+            Math.max(coords[1], current[3]),
+        ]
+    }, bbox);
+}
+
+// Because features come from tiled vector data, feature geometries may be split
+// or duplicated across tile boundaries. As a result, features may appear
+// multiple times in query results.
+// To prevent looping on duplicated features in computeMaxZoomFromGeoJsonFeatures we merge
+// bounding boxes of same key features
+function mergeBboxFromFeaturesWithSameKey(features) {
+    const uniqueIds = [];
+    const mergedBboxesArray = [];
+    for (const feature of features) {
+        // FIXME: supports only shapes for now
+        if (feature.geometry.type === 'Polygon') {
+            // Compute extent first
+            let bbox = VOID_BOUNDS;
+            const id = feature.properties['key'];
+            if (!uniqueIds.includes(id)) {
+                uniqueIds.push(id);
+                feature.geometry.coordinates.forEach(coordsPath => {
+                    bbox = computeBboxFromCoords(coordsPath, bbox);
+                });
+                mergedBboxesArray.push(bbox);
+            } else {
+                feature.geometry.coordinates.forEach(coordsPath => {
+                    bbox = computeBboxFromCoords(coordsPath, bbox);
+                });
+                let storedBbox = mergedBboxesArray[uniqueIds.indexOf(id)]
+                const mergedBbox = [
```

Comment:
Maybe this can be factored in a function? Looks similar to the one above.

Reply:
It's not really similar as here we only set the merged bbox value between two bboxes whereas the other is creating a bbox from coordinates.